### ISSUE TITLE
FinalizeHitChance and OverrideFinalHitChanceFns don't include enough information for FinalizeHitChance mods

### ISSUE THREAD

**Issue by Tedster59** [X2CommunityCore/X2WOTCCommunityHighlander#1368]
Issue #555 added these hooks, but they did not add information about the ability and target if available, such as in `X2AbilityToHitCalc_StandardAim`

![image](https://github.com/user-attachments/assets/2bb6f7bd-feb5-4b3a-b936-c1cc1999f727)

We should add new versions that pass through the ability and target information if available.

`GetAdditionalHitModifiers_CH` receives this information, but `FinalizeHitChance` does not.

Proposed Solution: Update `X2AbilityToHitCalc.FinalizeHitChance` to take in `XComGameState_Ability kAbility` and `AvailableTarget kTarget` as optional parameters, update `X2AbilityToHitCalc_StandardAim` to pass them, and then add a new delegate that will pass these through as well.

**Comment by Iridar** [comment]
That's gonna be problematic. `FinalizeHitChance` lives in `X2AbilityToHitCalc.uc`. 

The best we could do is create a `FinalizeHitChance_CH` that will be get called by `GetHitChance()`, and then wrap the original `FinalizeHitChance` in it, but that sounds like something that can very easily break if not handled properly by subclasses, especially mod-added ones. 

It will also start creeping towards a backwards compatibility hell, because original OverrideHitChanceFns will still need to be supported (which were a terrible implementation, in my opinion).

**Comment by Tedster59** [comment]
Yeah, that sounds like a compatibility nightmare.  I wonder what other mods are using this beyond LWOTC (and my standalone implementation), which is why I'd want to make sure the new stuff is optional and mods using the new hook can't expect the new additions to be passed in 100% of the time.

**Comment by Tedster59** [comment]
As an alternative implementation, would it be possible to trigger a new event here that passes a tuple with the shotbreakdown, ability, target (or just pass them in as event parameters) along with a bool for CHL to know whether a mod used it or not? This seems less robust than the delegate implementation, but would likely be better for backwards compatibility as none of the previous functionality is touched.

e.g from this:
![image](https://github.com/user-attachments/assets/03614631-fe7d-4439-8187-92d41f569961)

to something like this.

![image](https://github.com/user-attachments/assets/3ed40a2e-61b2-4d0c-954e-64db97643c19)

Any mod using this would have to deal with XModBase jank, but they can get around it by simply requiring my XMB Redux as a dependency or shipping it themselves in usual XMB style.

**Comment by Iridar** [comment]
No, GetHitChance is called by UI code, we can't trigger events here.

**Comment by Tedster59** [comment]
welp, so much for that plan then.
